===========================
How to Use 0-add_integer.py
===========================

Integer addition function ``add_integer(a, b=98)``.

Usage
=====

Importing the function from the module:
    >>> add_integer = __import__("0-add_integer").add_integer

Checking for module docstring:
    >>> d = __import__("0-add_integer").__doc__
    >>> len(d) > 1
    True

Checking for function docstring:
        >>> s = __import__("0-add_integer").add_integer.__doc__
        >>> len(s) > 1
        True

Regular Integers
::
    >>> add_integer(5, 10)
    15
    
    >>> add_integer(3, 3)
    6

negative integers
::   
    >>> add_integer(-3, -4)
    -7
    
    >>> add_integer(-8, 8)
    0

::
    >>> [add_integer(i, i ** 3) for i in range(8)]
    [0, 2, 10, 30, 68, 130, 222, 350]

floating values
:: 
    >>> add_integer(3.3, 7)
    10
    
    >>> add_integer(5, 7.8)
    12
    
    >>> add_integer(2.45, 3.1)
    5
    
    >>> add_integer(1, float(nan))
    Traceback (most recent call last):
    ...
    NameError: name 'nan' is not defined
    
    >>> add_integer(1, float(inf))
    Traceback (most recent call last):
    ...
    NameError: name 'inf' is not defined

    >>> add_integer(999999999999999999999999999999, 1)
    1000000000000000000000000000000

    >>> add_integer([1], 2)
    Traceback (most recent call last):
    TypeError: a must be an integer
    
    >>> add_integer(3, "2")
    Traceback (most recent call last):
    TypeError: b must be an integer
    
    >>> add_integer((2,), {"key": "value"})
    Traceback (most recent call last):
    TypeError: a must be an integer
    
    >>> add_integer(True, 1)
    Traceback (most recent call last):
    TypeError: a must be an integer
    
    >>> add_integer(0, False)
    Traceback (most recent call last):
    TypeError: b must be an integer
    
    >>> add_integer()
    Traceback (most recent call last):
    TypeError: add_integer() missing 1 required positional argument: 'a'
    
    >>> add_integer(5)
    103
    
    >>> add_integer(3, 5, 6)
    Traceback (most recent call last):
    TypeError: add_integer() takes from 1 to 2 positional arguments but 3 were given
